src/game/ZoneManager.cxx

```diff
@@ -61,37 +59,36 @@ void ZoneManager::spawn()
       LOG(LOG_ERROR) << "Something is wrong with a zone tile - Report this to the Cytopia Team!";
       continue;
     }
+
     // get the right zone enum of the zone tile
-    for (auto& zone : node->getTileData(Layer::ZONE)->zones)
+    std::vector<std::string> availableZoneTiles;
+    for (auto &zone : node->getTileData(Layer::ZONE)->zones)
     {
-      if (TileManager::instance().getTileIDsOfCategory(zone).empty())
+      auto &zoneTiles = TileManager::instance().getTileIDsOfCategory(zone);
+
+      if (zoneTiles.empty())
       {
         LOG(LOG_ERROR) << "No buildings available for zone: " << zone;
         continue;
       }
       else
       {
         // add all the tileIDs that are available for this zone to our vector
-        for (const auto &ID : TileManager::instance().getTileIDsOfCategory(zone))
-        {
-          myVec.emplace_back(ID);
-        }
+        availableZoneTiles.insert(availableZoneTiles.end(), zoneTiles.begin(), zoneTiles.end());
       }
     }
 
-    if (myVec.empty())
+    if (availableZoneTiles.empty())
     {
       LOG(LOG_ERROR) << "No buildings available! Aborting!";
       break;
     }
+
     // pick a random tileID from the vector
     // TODO: Check how many neighboring tiles are available and pick a building with correct tilesize
-    int randomIndex = rand() % myVec.size();
-
-    std::string building = myVec[randomIndex];
+    std::string building = *randomizer.choose(availableZoneTiles.begin(), availableZoneTiles.end());
```

Comment:
Does this address the TODO on line 88? If it does removing the TODO would be helpful. If it doesn't, then feel free to ignore this comment 😅